pkg/codec/config_test.go

```diff
@@ -1,17 +1,90 @@
 package codec_test
 
 import (
+	"bytes"
 	"encoding/json"
 	"reflect"
 	"testing"
 	"time"
 
+	"github.com/fxamacker/cbor/v2"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/smartcontractkit/chainlink-common/pkg/codec"
 )
 
+func TestNumberJSONMarshal(t *testing.T) {
+	type A struct {
+		B codec.Number
+	}
+
```

Comment:
Nothing in the test verifies that the number is encoded as a number.  This would pass if you didn't override the marshal/unmarshal.